ngraph/core/src/op/shuffle_channels.cpp

```diff
@@ -137,3 +142,54 @@ Shape op::ShuffleChannels::get_pre_shuffle_shape(const Shape& data_shape) const
 
     return res;
 }
+
+bool op::ShuffleChannels::evaluate(const HostTensorVector& outputs,
+                                   const HostTensorVector& inputs) const
+{
+    const auto arg = inputs[0]->get_data_ptr<const char>();
+    auto out = outputs[0]->get_data_ptr<char>();
+    Shape data_shape = inputs[0]->get_shape();
+    const Shape& ds = data_shape;
+    size_t elem_size = inputs[0]->get_element_type().size();
+
+    Shape pre_reshape_shape(4, 1);
+    size_t axis_zb = m_axis >= 0 ? m_axis : m_axis + data_shape.size();
+    for (size_t i = 0; i < axis_zb; ++i)
+    {
+        pre_reshape_shape[0] *= ds[i];
+    }
+
+    pre_reshape_shape[1] = m_group;
+    pre_reshape_shape[2] = ds[axis_zb] / m_group;
+
+    for (size_t i = axis_zb + 1; i < ds.size(); ++i)
+    {
+        pre_reshape_shape[3] *= ds[i];
+    }
+    AxisVector axes_order(data_shape.size());
+    std::iota(axes_order.begin(), axes_order.end(), 0);
+    size_t data_size = shape_size(data_shape) * elem_size;
+    std::vector<char> reshaped(data_size);
+    runtime::opt_kernel::reshape(
```

Comment:
do we really need this reshape? does it cost us smth? it doesn't move data -- tensor before and after would be the same. why we use it here and all the other channel/batch moving ops?

Reply:
looks like it makes sense since the target reshape shape already precalculated and it's a room for slight optimization.

Reply:
Removed first and last reshapes and added comments for clarity that they are intentionaly skipped